### PR TITLE
Add support for metadata embedding in image files

### PR DESCRIPTION
**Pull request by patriceac**
Extending task setting output format: 'txt', 'json', or 'embed' If 'embed', the metadata will be embedded in PNG files in tEXt chunks, and as EXIF UserComment for JPEG files. Both can then be read back from the image using vanilla JavaScript.

This is the backend change. More updates on the way for the UI part, but this is a prerequisite. Also undoes PR 7 (Allow the UI to disable file saving), which I will add back later with a cleaner implementation.

image

### CODE REVIEW

File: sdkit/utils/file_utils.py [hunk 1]
```diff
@@ -34,23 +36,39 @@ def save_images(images: list, dir_path: str, file_name='image', output_format='J
...
+                embedded_metadata.add_text(key, str(val))
+            targetImage.save(f'{path}.{file_format.lower()}', pnginfo=embedded_metadata)
+            return
+        elif output_format.lower() == 'embed' and file_format.lower() == 'jpeg':
+            targetImage = Image.open(f'{path}.{file_format.lower()}')
+            user_comment = json.dumps(metadata)
+            exif_data = targetImage.getexif()
+            exif_data[0x9286] = user_comment
+            targetImage.save(f'{path}.{file_format.lower()}', exif=exif_data)
+            return
```
Comment (madrang) [inline sdkit/utils/file_utils.py:70]: entries can be a list of dictionaries. Calling return here would only save the first item of that list.

Comment (madrang) [reply sdkit/utils/file_utils.py:70]: Validate using parallel renders. Only the first file will have the metadata.

**Comment by patriceac** [comment]
Will create a proper PR. Thanks Madrang for catching the issue.

**Comment by AkoZ** [comment]
Hey ... thanks for this .png matadata .. but not so easy to get in.
* so .. is it an implemented plugin ui ?
* then i'll re-read all the stuff and part python program to integrate it .. but may be as it will be done one day, i prefer you do it as a real dev plugin ui .. cheers !

**Comment by patriceac** [comment]
This PR was for the metadata _writing_ part. It has been superseded by https://github.com/easydiffusion/sdkit/pull/9. Please note this PR have not been merged just yet. Right now the priority is to stabilize the current Beta release to release it to our Main branch, so it may be a little while before we get to it.

The metadata reading part has not been released yet, but should be there soon, although NOT as part of sdkit. It will be a UI change in https://github.com/cmdr2/stable-diffusion-ui/tree/beta.

If you're looking to read themetadata in Python, here's how to do it:

    from PIL import Image

    img = Image.open("example.png")
    text_chunks = img.text
    for key, value in text_chunks.items():
        print(key + ": " + value)

Hope this helps.

**Pull request closed without merging**